### PR TITLE
Remove default qubit jax

### PR DESCRIPTION
**Pull request by albi3ro**
[sc-72785]

Yet another device removal. Code cov failures are in default.qubit.legacy and QubitDevice, so can be ignored.

### CODE REVIEW

File: tests/gradients/core/test_hadamard_gradient.py [hunk 1]
```diff
@@ -1156,7 +1156,7 @@ def cost_p(x):
         assert np.allclose(res_hadamard[1].detach(), res_param_shift[1].detach())
 
     @pytest.mark.jax
-    @pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
```
Comment (mudit2812) [inline tests/gradients/core/test_hadamard_gradient.py:1159]: We don't need to parametrize this test anymore. We can also remove `execute_fn` from the body of the test.

File: tests/gradients/core/test_jvp.py [hunk 2]
```diff
@@ -761,7 +761,7 @@ def cost_fn(params, tangent):
     # Include batch_dim!=None cases once #4462 is resolved
     @pytest.mark.jax
     @pytest.mark.parametrize("batch_dim", [None])  # , 1, 3])
-    @pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
+    @pytest.mark.parametrize("dev_name", ["default.qubit"])
```
Comment (mudit2812) [inline tests/gradients/core/test_jvp.py:764]: Same comment as above here

File: tests/gradients/core/test_pulse_gradient.py [hunk 3]
```diff
@@ -846,7 +846,7 @@ def test_raises_for_invalid_reorder_fn(self, reorder_fn):
 
 
 @pytest.mark.jax
-@pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
+@pytest.mark.parametrize("dev_name", ["default.qubit"])
```
Comment (mudit2812) [inline tests/gradients/core/test_pulse_gradient.py:849]: Same comment about removing the parametrization and any test function variables conditionally set based on device name.

File: tests/gradients/core/test_pulse_odegen.py [hunk 4]
```diff
@@ -997,7 +997,7 @@ def test_all_zero_diff_methods_multiple_returns_tape(self):
 
 
 @pytest.mark.jax
-@pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
+@pytest.mark.parametrize("dev_name", ["default.qubit"])
```
Comment (mudit2812) [inline tests/gradients/core/test_pulse_odegen.py:1000]: Same comment here

File: tests/gradients/core/test_vjp.py [hunk 5]
```diff
@@ -492,7 +492,7 @@ def test_tf(self, dev_name, tol):
 
     @pytest.mark.jax
     @pytest.mark.slow
-    @pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
+    @pytest.mark.parametrize("dev_name", ["default.qubit"])
```
Comment (mudit2812) [inline tests/gradients/core/test_vjp.py:495]: Same here

File: tests/gradients/finite_diff/test_finite_difference.py [hunk 6]
```diff
@@ -1025,7 +1025,7 @@ def cost_fn(params):
         assert np.allclose(hess[1].detach().numpy(), expected[1], atol=tol, rtol=0)
 
     @pytest.mark.jax
-    @pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
+    @pytest.mark.parametrize("dev_name", ["default.qubit"])
```
Comment (mudit2812) [inline tests/gradients/finite_diff/test_finite_difference.py:1028]: Same here

File: tests/gradients/finite_diff/test_finite_difference_shot_vec.py [hunk 7]
```diff
@@ -1052,7 +1052,7 @@ def cost_fn(params):
         assert np.allclose(hess[1].detach().numpy(), expected[1], atol=0.3, rtol=0)
 
     @pytest.mark.jax
-    @pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
+    @pytest.mark.parametrize("dev_name", ["default.qubit"])
```
Comment (mudit2812) [inline tests/gradients/finite_diff/test_finite_difference_shot_vec.py:1055]: Same here

File: tests/gradients/finite_diff/test_spsa_gradient.py [hunk 8]
```diff
@@ -1180,7 +1180,7 @@ def cost_fn(params):
         assert np.allclose(hess[1].detach().numpy(), expected[1], atol=atol, rtol=0)
 
     @pytest.mark.jax
-    @pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
+    @pytest.mark.parametrize("dev_name", ["default.qubit"])
```
Comment (mudit2812) [inline tests/gradients/finite_diff/test_spsa_gradient.py:1183]: Same here

File: tests/gradients/finite_diff/test_spsa_gradient_shot_vec.py [hunk 9]
```diff
@@ -1157,7 +1157,7 @@ def cost_fn(params):
         assert np.allclose(hess[1].detach().numpy(), expected[1], atol=spsa_shot_vec_tol, rtol=0)
 
     @pytest.mark.jax
-    @pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
+    @pytest.mark.parametrize("dev_name", ["default.qubit"])
```
Comment (mudit2812) [inline tests/gradients/finite_diff/test_spsa_gradient_shot_vec.py:1160]: Same here

File: tests/gradients/parameter_shift/test_parameter_shift.py [hunk 10]
```diff
@@ -2889,7 +2889,7 @@ def test_variance_gradients_agree_finite_differences(self, tol):
                 assert np.allclose(g, grad_F2[idx1][idx2], atol=tol, rtol=0)
 
     @pytest.mark.jax
-    @pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
+    @pytest.mark.parametrize("dev_name", ["default.qubit"])
```
Comment (mudit2812) [inline tests/gradients/parameter_shift/test_parameter_shift.py:2892]: Same here

File: tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py [hunk 11]
```diff
@@ -2367,7 +2367,7 @@ def test_torch(self, dev_name, broadcast, tol):
         assert np.allclose(hess[2][:, -1], np.zeros([2, 1, 1]), atol=tol, rtol=0)
 
     @pytest.mark.jax
-    @pytest.mark.parametrize("dev_name", ["default.qubit", "default.qubit.jax"])
+    @pytest.mark.parametrize("dev_name", ["default.qubit"])
```
Comment (mudit2812) [inline tests/gradients/parameter_shift/test_parameter_shift_shot_vec.py:2370]: Same here

**Review by mudit2812 — CHANGES_REQUESTED** [review]
Looks great, just one (very repetitive, sorry) comment about test parametrization. Otherwise happy to approve.

Let me know if you think it makes sense to leave in the test parametrization in case we want to include any future devices. However, the comment about not needing to conditionally set test function variables based on device name should be addressed regardless.

**Review by astralcai — APPROVED** [review]

**Review by mudit2812 — APPROVED** [review]
Looks good! Sorry for the tedium 😅
